src/handler.c

```diff
@@ -671,13 +671,21 @@ struct char_data *get_char_num(mob_rnum nr)
 /* put an object in a room */
 void obj_to_room(struct obj_data *object, room_rnum room)
 {
-  if (!object || room == NOWHERE || room > top_of_world)
+  if (!object || room == NOWHERE || room > top_of_world){
     log("SYSERR: Illegal value(s) passed to obj_to_room. (Room #%d/%d, obj %p)",
 	room, top_of_world, (void *)object);
+  }
   else {
-    object->next_content = world[room].contents;
-    world[room].contents = object;
-    IN_ROOM(object) = room;
+    if (world[room].contents == NULL){  // if list is empty
+      world[room].contents = object; // add object to list
+    }
+    else {
+      struct obj_data *i = world[room].contents; // define a temporary pointer
+      while (i->next_content != NULL) i = i->next_content; // find the first without a next_content
+        i->next_content = object; // add object at the end
```

Comment:
this will result in decreasing performance the more items are dropped into a room, allowing a bad actor (i.e. jerk player) to lag the game intentionally, to the point of crashing it due to checkpoint fail (I'm assuming tbamud still has 10 second checkpointing to guard against issues).

You'd be much better served using a double-linked list, instead of trying to append to the end of a traditional one, or at the very least add an additional pointer to the last node on a list that's updated when adding or removing from a room.

Reply:
I suspect the best solution would simply invert the list of objects in the room before they're displayed to the character, using a for loop. Unfortunately, my C skills are rather lacking and tackling that issue is beyond me at present. At some point, I might get around to figuring it out. Currently, I'm solving much simpler issues.

I haven't searched through the code yet, but I suspect there's a max limit to the number of objects that can be dropped in a room. If not, there needs to be! While that doesn't solve decreased performance, it might prevent a bad actor from causing a crash. I'll have to look into that. 

The MUD I used to play (heavily modified CircleMUD) had issues when 100 or more objects were dropped in a single room.

Reply:
They probably had issues and had to implement the 100 item limit.  Which I would consider a hack.  You are correct that this is a display issue you should customize yourself.

I guess my point is, this is a change that introduces consequences that aren't fully understood by everyone, and while you should feel free to modify your copy of the code as you see fit, this isn't the kind of change that should be part of the base game.  But that's just my opinion.

Reply:
Point taken. Although I'm not certain if you're aware, this order of objects in room issue started on the tbaMUD forums. Thomas thought it was a good catch and suggested a pull request. Thomas also authored the code snippet. I just pasted it in. 

I love the idea of continued development of the codebase. I have an abundance of ideas. Many of which, might ruffle some feathers or be met with resistance and denial. That's perfectly fine. As you suggest, I can implement them in my copy of the code.

I have no idea how progressive the maintainers are or how attached they might be to all the outdated legacy code, antiquated game design concepts, and methods of doing things. That's a dicussion I should probably have with the maintainers. In the mean time, I'm just throwing it out there to see what sticks. The response will give me feel for what maintainers and developers deem acceptable or unnacceptable.

Reply:
Fair enough, I don't follow the forums, just the pull requests, so I forget that I miss out on context sometimes.

I still think performance here will be a long term issue.  Maybe I'll whip something up.

Reply:
Man, if you do whip something up, that would be great! It's rather exciting to see the codebase move forward and I'd love to see a better solution. That would be greatly appreciated. The more source code I read, the more I learn and understand.